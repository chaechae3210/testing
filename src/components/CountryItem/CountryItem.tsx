import Image from "next/image";
import style from "./CountryItem.module.css";
import { useRouter } from "next/router";

type TProps = {
  code: string;
  commonName: string;
  flagEmoji: string;
  flagImg: string;
  population: number;
  region: string;
  capital: string[];
};

export default function CountryItem({
  code,
  commonName,
  flagEmoji,
  flagImg,
  population,
  region,
  capital,
}: TProps) {
  const router = useRouter();

  const onClickItem = () => {
    router.push(`/country/${code}`);
  };

  return (
    <div onClick={onClickItem} className={style.container}>
      <div className={style.flag_img}>
        <Image src={flagImg} alt="" fill />
      </div>
      <div className={style.content}>
        <div className={style.name}>
          {flagEmoji} {commonName}
        </div>
        <div>지역 : {region}</div>
        <div>수도 : {capital.join(", ")}</div>
        <div>인구 : {population}</div>
      </div>
    </div>
  );
}
