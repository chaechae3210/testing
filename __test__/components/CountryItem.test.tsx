import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import CountryItem from "@/components/CountryItem/CountryItem";
import mockRouter from "next-router-mock";

jest.mock("next/router", () => jest.requireActual("next-router-mock"));

const mockProps = {
  code: "KOR",
  commonName: "South Korea",
  flagEmoji: "🇰🇷",
  flagImg: "https://flagcdn.com/w320/kr.png",
  capital: ["Seoul"],
  region: "Asia",
  population: 51780579,
};

describe("CountryItem Component", () => {
  it("CountryItem에 프롭스를 넘겼을 때 국가 정보가 올바르게 렌더링 됨.", () => {
    render(<CountryItem {...mockProps} />);

    expect(screen.getByText("🇰🇷 South Korea")).toBeInTheDocument();
    expect(screen.getByText("지역 : Asia")).toBeInTheDocument();
    expect(screen.getByText("수도 : Seoul")).toBeInTheDocument();
    expect(screen.getByText("인구 : 51780579")).toBeInTheDocument();
  });

  // 위의 테스트를 스냅샷 테스트로 바꾸었을 때
  it("CountryItem 스냅샷 테스트", () => {
    const { container } = render(<CountryItem {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it("li를 클릭하면 국가 정보 디테일 페이지로 push됨", async () => {
    mockRouter.push("/");

    render(<CountryItem {...mockProps} />);
    const container = screen.getByRole("listitem");
    await userEvent.click(container);

    expect(mockRouter.asPath).toEqual(`/country/${mockProps.code}`);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
