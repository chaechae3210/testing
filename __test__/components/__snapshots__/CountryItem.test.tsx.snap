// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryItem Component CountryItem 스냅샷 테스트 1`] = `
<div>
  <li
    class="container"
  >
    <div
      class="flag_img"
    >
      <img
        alt=""
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="(max-width: 320px)"
        src="/_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
      />
    </div>
    <div
      class="content"
    >
      <div
        class="name"
      >
        🇰🇷
         
        South Korea
      </div>
      <div>
        지역 : 
        Asia
      </div>
      <div>
        수도 : 
        Seoul
      </div>
      <div>
        인구 : 
        51780579
      </div>
    </div>
  </li>
</div>
`;

exports[`CountryItem Component renders homepage unchanged 1`] = `
<div>
  <li
    class="container"
  >
    <div
      class="flag_img"
    >
      <img
        alt=""
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="(max-width: 320px)"
        src="/_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fflagcdn.com%2Fw320%2Fkr.png&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
      />
    </div>
    <div
      class="content"
    >
      <div
        class="name"
      >
        🇰🇷
         
        South Korea
      </div>
      <div>
        지역 : 
        Asia
      </div>
      <div>
        수도 : 
        Seoul
      </div>
      <div>
        인구 : 
        51780579
      </div>
    </div>
  </li>
</div>
`;
